File: Array/1029. Two City Scheduling/Solution.java
class Solution {
    public int twoCitySchedCost(int[][] costs) {
        int N=costs.length;
        Arrays.sort(costs, new Comparator<int[]>(){
            @Override
            public int compare(int[] a, int[] b){
                return (a[0]-a[1])<(b[0]-b[1])?-1:1;
            }
        });
        int res=0;
        for (int i=0; i<N/2; i++){
            res+=costs[i][0];
        }
        for (int i=N/2; i<N; i++){
            res+=costs[i][1];
        }
        return res;
    }
}

File: Array/1029. Two City Scheduling/Solution.py
class Solution:
    # divide all peole in to 2 groups with LOWEST cost
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        N=len(costs)
        costs.sort(key=lambda x:x[0]-x[1])
        res=0
        for cost in costs[:N//2]:
            res+=cost[0]
        for cost in costs[N//2::]:
            res+=cost[1]
        return res